osquery/tables/system/darwin/cups_jobs.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <cups/cups.h>
+
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+class SafeCupsJobs {
+ public:
+  cups_job_t* job_list;
+  int num_jobs;
+
+  SafeCupsJobs() {
+    num_jobs = cupsGetJobs(&job_list, nullptr, 0, CUPS_WHICHJOBS_ALL);
+  }
+
+  ~SafeCupsJobs() {
+    cupsFreeJobs(num_jobs, job_list);
+  }
+};
+
+QueryData genCupsJobs(QueryContext& request) {
+  QueryData results;
+  SafeCupsJobs jobs;
+
+  for (decltype(jobs.num_jobs) i = 0; i < jobs.num_jobs; ++i) {
+    Row r;
+    r["title"] = std::string(jobs.job_list[i].title);
```

Comment:
Not super important, but here and above, any reason for using `std::string` over `SQL_TEXT`? Seems like it would fit better here given the `INTEGER` uses below.